src/systems/thruster/Thruster.cc

```diff
@@ -333,6 +396,17 @@ double ThrusterPrivateData::ThrustToAngularVec(double _thrust)
   return propAngularVelocity;
 }
 
+/////////////////////////////////////////////////
+void ThrusterPrivateData::UpdateThrustCoefficient()
+{
+  double calculatedThrustCoefficient = this->alpha_1 + this->alpha_2 *
+      (((1 - this->wakeFraction) * this->linearVelocity)
+      / (this->propellerAngVel * this->propellerDiameter));
+  if (!std::isnan(calculatedThrustCoefficient)
+      && calculatedThrustCoefficient!=0)
```

Comment:
@marcoag meta: hmm, something's off here. If the thrust coefficient is zero, then no thrust should apply. If the propeller angular velocity is zero, then no thrust should apply and the thrust coefficient is not defined (and it doesn't matter anyways). I see this cause-effect relationship is reversed in this plugin: we assume thrust is always applicable and compute propeller angular velocity. I think we should set that straight. Then you would not need this conditional.

Reply:
When the plugin starts the angular velocity is zero, making the thrust coefficient nan. Since it's used later on to calculate the propeller angular velocity then the model would never start moving. The intent of this check was to break this loop and use the default thrust coefficient which is 1 and won't affect the final result. I think it might suffice with the `isnan` check for this or another option is to only update the thrust coefficient once the propeller angular velocity is not 0, maybe the later makes more sense?

Reply:
Ah, almost forgot to reply. 

I guess my point is that the current implementation is not a good approximation. It takes thrust commands instead of propeller angular velocities, and applies forces instantaneously. That's why we are running into this odd corner cases. It makes me wonder, why use this improved thrust coefficient if the dynamics are not there.

Anyways, I think that if we want to keep everything as-is, we could (a) set propeller angular velocity to zero if thrust is zero, bypassing all calculations, and (b) solve the quadratic-like equation to get propeller angular velocity in all other cases, as if it had reached steady state instantaneously (like with forces).

Reply:
I don't know the reason to take thrust commands instead of propeller angular velocities here, the later would be even easier since it seems the thrust gets converter into an angular velocity anyway.

a) The propeller angular velocity would be zero if thrust is zero using calculations, so there doesn't seem a to be a need to bypass them. 

Still the problem is a chicken egg dependency between velocity and thrust coefficient, due to a mutual dependency, if any of them is zero it would make the other one zero, preventing the vehicle to move when starting from stationary. With the conditional above I was trying to avoid this problem. I guess it makes more sense to only calculate when `angular_velocity != 0`  and use the default thrust coefficient (or latest) when that happens. Will add this change for now.

Reply:
As per offline discussin, let's go with this as-is. I ran a couple, very much simplified numerical experiments and the coefficient converges even if we start from a guess. I can't swear on convergence, but if the sim looks OK we should be fine.

> solve the quadratic-like equation to get propeller angular velocity in all other cases, as if it had reached steady state instantaneously (like with forces).

Re: to myself. This is not super easy, I missed the linear velocity term in the thrust coefficient model.